lanelet2/lanelet2_routing/src/Route.cpp

```diff
@@ -132,7 +133,17 @@ Route::~Route() noexcept = default;
 Route& Route::operator=(Route&& other) noexcept = default;
 Route::Route(Route&& other) noexcept = default;
 Route::Route(LaneletPath shortestPath, std::unique_ptr<RouteGraph> graph, LaneletSubmapConstPtr laneletSubmap) noexcept
-    : graph_{std::move(graph)}, shortestPath_{std::move(shortestPath)}, laneletSubmap_{std::move(laneletSubmap)} {}
+    : graph_{std::move(graph)}, shortestPath_{std::move(shortestPath)}, laneletSubmap_{std::move(laneletSubmap)} {
+    
+    if (shortestPath_.size() != 0 && shortestPath_.back().leftBound3d().size() != 0 && shortestPath_.back().rightBound3d().size() != 0)
+    {      
+      auto left_pt_last = shortestPath_.back().leftBound3d().back();
+      auto right_pt_last = shortestPath_.back().rightBound3d().back();
+      
+      lanelet::Point3d end_point {lanelet::utils::getId(), (left_pt_last.x() + right_pt_last.x())/2, (left_pt_last.y() + right_pt_last.y())/2,(left_pt_last.z() + right_pt_last.z())/2};
```

Comment:
It would probably make more sense to just call centerline() on the shortestPath_.back() lanelet and take the last point. Having this point deviate from that point might cause some unexpected behavior.

Reply:
using centerline now